elements/pl-multiple-choice/pl-multiple-choice.py

```diff
@@ -85,52 +101,52 @@ def prepare(element_html, data):
     len_incorrect = len(incorrect_answers)
     len_total = len_correct + len_incorrect
 
-    enable_nota = pl.get_boolean_attrib(element, 'none-of-the-above', NONE_OF_THE_ABOVE_DEFAULT)
-    enable_aota = pl.get_boolean_attrib(element, 'all-of-the-above', ALL_OF_THE_ABOVE_DEFAULT)
+    nota = get_nota_aota_attrib(element, 'none-of-the-above', NONE_OF_THE_ABOVE_DEFAULT)
+    aota = get_nota_aota_attrib(element, 'all-of-the-above', ALL_OF_THE_ABOVE_DEFAULT)
 
-    nota_correct = False
-    aota_correct = False
-    if enable_nota or enable_aota:
-        prob_space = len_correct + enable_nota + enable_aota
-        rand_int = random.randint(1, prob_space)
-        # Either 'None of the above' or 'All of the above' is correct
-        # with probability 1/(number_correct + enable-nota + enable-aota).
-        # However, if len_correct is 0, nota_correct is guaranteed to be True.
-        # Thus, if no correct option is provided, 'None of the above' will always
+    if aota in ['correct', 'random'] and len_correct < 2:
+        # To prevent confusion on the client side
+        raise Exception('pl-multiple-choice element must have at least 2 correct answers when all-of-the-above is set to "correct" or "random"')
+
+    if len_correct == 0:
+        if nota == 'false':
+            # This means the code needs to handle the special case when len_correct == 0
```

Comment:
I don't understand what this comment is saying - is there other code in this element that handles this special case?

Reply:
This was indeed confusing. I separated the logic to a separate check before this point.